internal/querycoordv2/task/merger.go

```diff
@@ -128,12 +130,22 @@ func (merger *Merger[K, R]) mergeQueue(id K, queue chan MergeableTask[K, R]) {
 	defer merger.processors.Remove(id)
 
 	len := len(queue)
-	task := <-queue
-	for i := 1; i < len; i++ {
-		task.Merge(<-queue)
+	var task MergeableTask[K, R]
+	// output every per maxSegmentsPerTask
+	for i, curr := 0, 0; i < len; i, curr = i+1, (curr+1)%maxSegmentsPerTask {
+		if curr == 0 {
+			if task != nil {
+				merger.outCh <- task
```

Comment:
have to set `task = nil` after line 138, or the last task will be output again at line 146